src/main/java/org/spongepowered/common/mixin/core/entity/MixinArmorEquipable.java

```diff
@@ -35,86 +35,75 @@
 import net.minecraft.util.EnumHand;
 import org.spongepowered.api.data.type.HandType;
 import org.spongepowered.api.entity.ArmorEquipable;
+import org.spongepowered.api.entity.Equipable;
 import org.spongepowered.api.item.ItemTypes;
 import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.equipment.EquipmentType;
 import org.spongepowered.asm.mixin.Implements;
 import org.spongepowered.asm.mixin.Interface;
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.common.data.type.SpongeEquipmentType;
 import org.spongepowered.common.entity.living.human.EntityHuman;
+import org.spongepowered.common.item.inventory.util.ItemStackUtil;
 
 import java.util.Optional;
 
 import javax.annotation.Nullable;
 
-// All implementors of ArmorEquipable
+// All living implementors of ArmorEquipable
 @Mixin({EntityArmorStand.class, EntityGiantZombie.class, EntityPlayerMP.class, AbstractSkeleton.class, EntityZombie.class, EntityHuman.class})
 @Implements(@Interface(iface = ArmorEquipable.class, prefix = "equipable$"))
-public abstract class MixinArmorEquipable extends MixinEntityLivingBase {
+public abstract class MixinArmorEquipable extends MixinEntityLivingBase implements Equipable {
 
-    public Optional<ItemStack> equipable$getHelmet() {
-        @Nullable final net.minecraft.item.ItemStack itemStack = this.getItemStackFromSlot(EntityEquipmentSlot.HEAD);
-        return Optional.ofNullable(itemStack.isEmpty() ? null : (ItemStack) itemStack.copy());
+    @Override
+    public boolean canEquip(EquipmentType type) {
+        return true;
     }
 
-    public void equipable$setHelmet(ItemStack helmet) {
```

Comment:
Why are these implementations being removed? Wouldn't it make sense to keep them around for slightly faster implementation?